index.js

```diff
@@ -3,54 +3,85 @@ let invalid = false
 
 
 const passwordPattern = {
-    charLength: function() {
-        if (password.value.length() >= 8){
+    charLength: function(password) {
+        if (password.length() >= 8){
             return valid
-        } else if (password.value.length() < 9){
+        } else if (password.length() < 8){
             return invalid
         }
     },
-    upCase: function() {
-        if (password.value.toUpperCase()) {
+    upCase: function(password) {
+        if (password.toUpperCase()) {
             return valid
-        }else if (!password.value.toUpperCase()){
+        }else if (password.value.toUpperCase() != password.value){
             return invalid
         }
     },
-    lowCase: function() {
+    lowCase: function(password) {
         if (password.value.toLowerCase()) {
             return valid
-        }else if (!password.value.toLowerCase()){
+        }else if (password.value.toLowerCase() != password.value){
             return invalid
         }
     
 },
-    num: function(){
-        if (password.value === [0-9]){
-            return valid
-        } else if (password.value != [0-9]){
-            return invalid
+    num: function(password){
+        let numbers = [0,1,2,3,4,5,6,7,8,9]
+        for (i = 0; i < numbers.length; i++){
+            if(password.indexof(password[i]) >= 0){
+                return valid
+            }
         }
     },
-    specialChar: function(){
-        let sc = /[ `!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?~]/;
-        if (password.value === sc){
+       
+    specialChar: function(password) {
+        let sc = ['!','@','#','$','%','^','&','?','~'] 
+          for (i = 0; i < sc.length; i++)
+        if (password.indexof(sc[i]) >= 0){
             return valid
-        }else if  (password.value != sc){
+        }else if  (password.value === !sc){
             return invalid
-        }
+        } 
     }
 }
+
+
 const validatePassword = (password) => {  
-      
- if (password === passwordPattern.specialChar && 
-    password === passwordPattern.num &&
-    password === passwordPattern.upCase &&
-    password === passwordPattern.lowCase &&
-    password === passwordPattern.charLength){
+ if (password === passwordPattern.num || 
+    password === passwordPattern.specialChar ||
```

Comment:
not sure exactly what the passwordPattern is is this something you found online?

Reply:
no, its just what I called the criteria the password had to follow.